packages/manager/src/features/CloudPulse/shared/CloudPulseResourcesSelect.tsx

```diff
@@ -20,62 +25,65 @@ export interface CloudPulseResourcesSelectProps {
 
 export const CloudPulseResourcesSelect = React.memo(
   (props: CloudPulseResourcesSelectProps) => {
-    const [selectedResource, setResources] = React.useState<
-      CloudPulseResources[]
-    >([]);
     const { data: resources, isLoading } = useResourcesQuery(
       props.region && props.resourceType ? true : false,
       props.resourceType,
       {},
       { region: props.region }
     );
 
+    const [selectedResources, setSelectedResources] = React.useState<
+      CloudPulseResources[]
+    >([]);
+
     const getResourcesList = (): CloudPulseResources[] => {
       return resources && resources.length > 0 ? resources : [];
     };
 
+    // Once the data is loaded, set the state variable with value stored in preferences
     React.useEffect(() => {
-      const defaultResources = resources?.filter((instance) =>
-        props.defaultSelection?.includes(instance.id)
-      );
+      const defaultResources = getUserPreferenceObject()?.resources;
+      if (resources) {
+        if (defaultResources) {
+          const resource = getResourcesList().filter((resource) =>
+            defaultResources.includes(String(resource.id))
+          );
 
-      if (defaultResources && defaultResources.length > 0) {
-        setResources(defaultResources);
-        props.handleResourcesSelection(defaultResources!);
+          props.handleResourcesSelection(resource);
+          setSelectedResources(resource);
+        } else {
+          setSelectedResources([]);
+          props.handleResourcesSelection([]);
+        }
+      } else {
+        setSelectedResources([]);
       }
-
       // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [resources, props.region]); // only on any resources or region change, select defaults if any
+    }, [resources, props.region, props.resourceType]);
 
     return (
       <Autocomplete
-        onChange={(_: any, resourceSelections: any) => {
-          setResources(resourceSelections);
+        onChange={(_: any, resourceSelections: CloudPulseResources[]) => {
+          updateGlobalFilterPreference({
+            [RESOURCES]: resourceSelections.map((resource: { id: any }) =>
+              String(resource.id)
+            ),
+          });
+          setSelectedResources(resourceSelections);
           props.handleResourcesSelection(resourceSelections);
         }}
         autoHighlight
         clearOnBlur
-        data-testid={'Resource-select'}
+        data-testid="Resource-select"
```

Comment:
```suggestion
        data-testid="resource-select"
```
Nit: consistency with capitalization

Reply:
done